eclipse-collections/src/main/java/org/eclipse/collections/impl/set/mutable/SetAdapter.java

```diff
@@ -152,6 +152,16 @@ public boolean containsAll(Collection<?> collection)
     @Override
     public boolean equals(Object obj)
     {
+        if (obj == this)
+        {
+            return true;
+        }
+
+        if (!(obj instanceof Set))
+        {
+            return false;
+        }
+
```

Comment:
This seems out of place.

Reply:
@motlin why is that? Maybe we don't want to test for being in instance of `Set` and let `AbstractCollectionAdapter` define `equals` directly?